### ISSUE TITLE
Compatibility with Android TV

### ISSUE THREAD

**Issue by Twanislas** [d4rken/wakelock-revamp#9]
Hi !

It would be very useful to add compatibility with Android TV so users could install the app from the play store on set top boxes instead of sideloading.

Thanks for your consideration :)

Context: https://github.com/JeffLIrion/python-androidtv/issues/18#issuecomment-478183332

**Comment by d4rken** [comment]
Playstore needs a few extra graphics for Android TV, I can look into that.

We would also need to make sure that the menu can be navigated by remote, I have not looked into how good that current works. Feel free to help out with that :).

**Comment by bieniu** [comment]
@d4rken Wakelock menu can be navigated by remote on Xiaomi Mi Box 3 (Android TV 9.0) without any problems. I checked yesterday.

**Comment by Twanislas** [comment]
+1 on what @bieniu said. I can confirm it works ok on a Mibox S (4) (ATV 9.0). That being said, it's not the perfect interface for ATV but since it's kinda "set it once and forget it" that's pretty ok.

**Comment by CreaaZ** [comment]
I can confirm it is running on Sony Bravia TV KD-55X8505C as intended. Navigation with remote is also fine - installed via ADB.

**Comment by magicse** [comment]
After last update (2604) of MiBox S(4)  (ATV 9.0), Wakelock not work and MiBox S puts in deep sleep.

**Comment by potatoxbe** [comment]
> After last update (2604) of MiBox S(4) (ATV 9.0), Wakelock not work and MiBox S stay in deep sleep.

Yes same issue here.

**Comment by keizie** [comment]
> After last update (2604) of MiBox S(4) (ATV 9.0), Wakelock not work and MiBox S stay in deep sleep.

FYI, USB Ethernet (connected to MiBox S) LEDs turns on and off soon, periodically.
iPhone can connect to MIBOX4 when the time is right.

**Comment by orholly** [comment]
> > After last update (2604) of MiBox S(4) (ATV 9.0), Wakelock not work and MiBox S stay in deep sleep.
> 
> Yes same issue here.

Me too.

**Comment by d4rken** [comment]
There is no specific change required in the app. This is likely a ROM change that increases restrictions on battery consumptions. Somewhere in the the system settings is likely a menu where you can exclude wakelock-revamp from these restrictions.

**Comment by magicse** [comment]
> 
> 
> There is no specific change required in the app. This is likely a ROM change that increases restrictions on battery consumptions. Somewhere in the the system settings is likely a menu where you can exclude wakelock-revamp from these restrictions.

Interesting moment - after restart of mi box S with installed and activated WakeLock , wakelock does everything right, and does not allow the device to go into deep sleep ... but it does it only once and after waking up, this does not happen anymore until the next reboot. It looks like the service is not restarting

**Comment by d4rken** [comment]
> It looks like the service is not restarting

Check whether the notification is visible. If it's visible, then the app is doing it's job. If the notification is not visible (and thus the service not started), then there might be an issue.

**Comment by magicse** [comment]
no any notifications....

**Comment by magicse** [comment]
Also I found this

**Comment by d4rken** [comment]
That could be an explanation... We can't modify the ROM and whitelist the package... :thinking:

**Comment by magicse** [comment]
Here is example
https://medium.com/@igor.stebliy/android-tv-application-notifications-on-o-c6820ada2086

**Comment by d4rken** [comment]
> You will need to have access to the AOSP sources to white-list your application and configure proper access to the system API.

**Comment by magicse** [comment]
ok

**Comment by saarsinai** [comment]
mi box s, latest android version (9), wakelock side loaded, gave all permission, load on startup, power optimization turned off, none made the mi box stay awake after pressing the power button

**Comment by arefin932** [comment]
I am getting "Ethernet not Connected" error every time my Xiaomi Mi Box S wakes up from sleep. I have to unplug and replug the USB Ethernet  adapter or reboot the box to make the USB Ethernet recognized. Wakelock was working fine until I upgraded my Box to Android 9 (2604). I have done factory reset on my device after the upgrade but no luck. I have "load on startup" checked in the wakelock with the following options, power optimization turned off for Wakelock on the Mibox. But still no luck.  
https://imgur.com/S0kHOY8

**Comment by magicse** [comment]
Powermanager has very aggressive save mode. And Wfi, CEC, USB, partial_wake_locks not work in sleep mode at all!
Thant is why Wifi (Chromecast) or phone app remote control or CEC can't wake up device from sleep mode.
On build 2231 was the same but I had posiblity use 3rd party software for partial wake lock and this helped me.
But now on 2604 it not work at all !!!!
Please fix Powermanager and partial_wake_locks!!!
Here link to bug reports [....](https://docs.google.com/forms/d/e/1FAIpQLSchnfHCQW6woj8XTB02Tw_B9JGuCeDB9MECon-iewehOtQVMA/viewform) write there about this problem

**Comment by VoyteckPL** [comment]
I have usb dongle remote control which works great when Mi Box S is turned on. I can turn it off via Usb remote but I can't turn it on back again. Only bluetooth remote can turn Mi Box S on.

**Comment by darkkatarsis** [comment]
Same here. After updating Mi box to android 9, wakelock stopped working. Someone has already found a workaround?

Ps. eth from usb not working either

**Comment by IgorEisberg** [comment]
Yep, same issue with Mi Box 3 after update to Android 9. Basically, rendered a universal wireless+IR remote useless, including Xiaomi's own Mi Remote app. (same issue as @VoyteckPL described, no matter what sent the IR signal to put the Mi Box to sleep)
You will also see in that broken sleep state that the remote is losing connection with the Mi Box completely, and the remote's LED will be flashing as if it's trying to establish a connection with the dongle to no avail.
Things I noticed:
1. If you enable USB debugging, go into this broken sleep state, then try to connect the Mi Box to a PC - the PC will report "USB device not recognized" and ADB will show no devices connected. Once you wake the device with the Bluetooth remote and reconnect the USB cable to the PC, the Mi Box is recognized on PC again.
2. If you enable USB debugging, connect the Mi Box to a PC **while the device is awake**, you will be able to use the IR remote normally with no issues at all - both going to sleep and waking up work normally until you disconnect the Mi Box from the PC.

**Comment by Cr0WD** [comment]
Any solution ?

**Comment by arefin932** [comment]
I ended us using the following USB ethernet adapter and it works
flawlessly.  Only a USB adapter with AX88179 chipset works flawlessly with
Android TV  OS and the uGreen one  has AX88179 chipset.  Hope it helps.

https://www.amazon.com/UGREEN-Ethernet-1000Mbps-Nintendo-Chromebook/dp/B00LLUEJFU/ref=sr_1_3?dchild=1&keywords=ax88179&qid=1601677211&s=electronics&sr=1-3

**Comment by darkkatarsis** [comment]
This is not a solution, if you have a connected USB adapter, you cannot use the ADB Debugger function.

**Comment by jfk88** [comment]
Same on MiTV-MSSP2 (Android 9) with battery optimisation disabled.

**Comment by Xitro01** [comment]
Would love to be able to install it from the store for my Mi TV Box, not a big fan of sideloading stuff. Hopefully someone can help out with this. I'm not a programmer myself, otherwise I would've helped in my spare time.

**Comment by ihrapsa** [comment]
> Would love to be able to install it from the store for my Mi TV Box, not a big fan of sideloading stuff. Hopefully someone can help out with this. I'm not a programmer myself, otherwise I would've helped in my spare time.

Unfortunately, the app is not working on Mi box anymore (or any other android tv I know of).

**Comment by joshuaspence** [comment]
> > Would love to be able to install it from the store for my Mi TV Box, not a big fan of sideloading stuff. Hopefully someone can help out with this. I'm not a programmer myself, otherwise I would've helped in my spare time.
> 
> Unfortunately, the app is not working on Mi box anymore (or any other android tv I know of).

It works for me on the latest version of AndroidTV on a Hitachi device.

**Comment by Xitro01** [comment]
> > Would love to be able to install it from the store for my Mi TV Box, not a big fan of sideloading stuff. Hopefully someone can help out with this. I'm not a programmer myself, otherwise I would've helped in my spare time.
> 
> Unfortunately, the app is not working on Mi box anymore (or any other android tv I know of).

That sucks, any alternatives available?

**Comment by magicse** [comment]
The alternative is to require an item in the settings menu. Ш
Some developers (for example, Xiaomi) simply blocked this item for no reason.
It is for this reason that I abandoned the Xiaomi Mi box S4 and switched to another manufacturer.

Note. The deep standby mode turning off more STB functions (Google services, messengers, Internet, etc.) and results in longer activation latency.
![screen_29](https://user-images.githubusercontent.com/13585785/185880365-ab2596e9-9d46-4933-aab4-a0e4b3079ba7.png)
![screen_30](https://user-images.githubusercontent.com/13585785/185880411-eb1e07f8-cf4b-4da1-99cf-4f851287f74e.png)

**Comment by lollilol** [comment]
Any update on this? So far the problem is the notification? Because it's handled differently in Android TV compared to normal Android?
